packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -455,3 +428,153 @@ function defaultGenerateClientInfo({ request }: GraphQLRequestContext) {
     return {};
   }
 }
+
+// Creates trace details from request variables, given a specification for modifying
+// values of private or sensitive variables.
+// The details include the variables in the request, TODO(helen): and the modifier type.
+// If sendVariableValues is {safelistAll: bool} or {exceptVariableNames: Array}, it will act similarly to
+// to the to-be-deprecated options.privateVariables, except that the redacted variable
+// names will still be visible in the UI when 'true.'
+// If sendVariableValues is null, we default to the safeListAll = false case.
+export function makeTraceDetails(
+  variables: Record<string, any>,
+  sendVariableValues?: VariableValueOptions,
+  operationString?: string,
+): Trace.Details {
+  const details = new Trace.Details();
+  const variablesToRecord = (() => {
+    if (sendVariableValues && 'valueModifier' in sendVariableValues) {
+      // Custom function to allow user to specify what variablesJson will look like
+      const originalKeys = Object.keys(variables);
+      const modifiedVariables = sendVariableValues.valueModifier({
+        variables: variables,
+        operationString: operationString,
+      });
+      return cleanModifiedVariables(originalKeys, modifiedVariables);
+    } else {
+      return variables;
+    }
+  })();
+
+  // Note: we explicitly do *not* include the details.rawQuery field. The
+  // Engine web app currently does nothing with this other than store it in
+  // the database and offer it up via its GraphQL API, and sending it means
+  // that using calculateSignature to hide sensitive data in the query
+  // string is ineffective.
+  Object.keys(variablesToRecord).forEach(name => {
+    if (
+      sendVariableValues == null ||
```

Comment:
Since the only valid TS values are objects, you can just check `sendVariableValues` without the `== null` comparison. (We do generally avoid `==` since it is subtle how it is different from `===`.)